File: pset1/cash.c
#include <cs50.h>
#include <stdio.h>

int calculate_coins(int cents);
int main(void)
{
    int cents;
    do
    {

        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    int Total = calculate_coins(cents);
    printf("%i\n", Total);
}

int calculate_coins(int cents)
{
    int quarters = cents / 25;
    cents %= 25;

    int dimes = cents / 10;
    cents %= 10;

    int nickels = cents / 5;
    cents %= 5;

    int pennies = cents / 1;

    return quarters + dimes + nickels + pennies;
}

File: pset6/cash.py
from cs50 import get_float

# Function to calculate coins


def calculate_coins(cents):
    quarters = cents // 25
    cents %= 25

    dimes = cents // 10
    cents %= 10

    nickels = cents // 5
    cents %= 5

    pennies = cents // 1

    return quarters + dimes + nickels + pennies


# Prompt user for valid input
while True:
    dollars = get_float("Change owed: ")
    if dollars >= 0:
        break

# Convert dollars to cents
cents = round(dollars * 100)

# Print result
total = calculate_coins(cents)
print(total)
